sql/range.go

```diff
@@ -0,0 +1,269 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+// Ranges are a collection of ranges that represent different (and potentially overlapping) filter expressions.
+type Ranges []Range
```

Comment:
nit: type names that are just a plural of another type are a bit of a headache to read

Reply:
That's fair! I debated calling them `RangeCollection` or just `Ranges`, I'll change the name